Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -38,41 +36,27 @@ void ChangePulsetime::init() {
 /** Execute the algorithm.
  */
 void ChangePulsetime::exec() {
-  EventWorkspace_const_sptr in_ws = getProperty("InputWorkspace");
+  EventWorkspace_sptr in_ws;
+  SpectrumIndexSet indexSet(0);
+
+  std::tie(in_ws, indexSet) = std::tuple<EventWorkspace_sptr, SpectrumIndexSet>(
```

Comment:
The only way I can think of to improve the situation is with explicit specialisation which increases the maintenance cost. (It might be necessary to specialise for each workspace type which is ugly).